nservicebus/azure-service-bus/performance-tuning_body_asb_[,7).partial.md

```diff
@@ -0,0 +1,14 @@
+### CPU vs IO bound processing
+
+The following settings are used in order to tune performance in Azure Service Bus version 6 and below:
+
+ - `MaximumConcurrencyLevel`
+ - `BatchSize`
+ - `LockDuration`
+ - `MaxDeliveryCount`
+
+In scenarios where handlers are CPU intense and have very little IO, it is advised to lower the number of threads to one and increase the `BatchSize`. `LockDuration` and `MaxDeliveryCount` might require an adjustment to match the batch size, taking into account the number of messages that end up in the dead letter queue.
+
+In scenario where handlers are IO intense, it is advised to set the number of threads  to 12 threads per logical core using `MaximumConcurrencyLevel` setting, and set the `BatchSize` to a number of messages that takes to process. Take into account the expected (or measured) processing time and IO latency of a single message. Start with a small `BatchSize` and through adjustment and measurement gradually increase it, while adjusting accordingly `LockDuration` and `MaxDeliveryCount`.
```

Comment:
In older versions of the CLR it would default to 25 threads per logical core, 12 is about half of that (12 for receiving, 12 for sending, 1 for GC) is the logic. In recent versions of the CLR this logic doesn't hold though as it now dynamically adjusts